src/libs/scene_utils/factory/scene_factory_impl.ts

```diff
@@ -0,0 +1,177 @@
+
+  private static readonly s3Client: S3Client = new S3Client();
+
+  private readonly deserializer: Deserializer = new Deserializer();
+  private readonly serializer: Serializer = new Serializer();
+
+  public async loadOrCreateSceneIfNotExists(workspaceId: string, sceneId: string): Promise<IotTwinMakerSceneImpl> {
+    return this.createScene(workspaceId, sceneId).catch((e) => {
+      if (e.code === 'ConflictException') {
+        console.log('Scene already exists, loading scene...');
+        return this.loadScene(workspaceId, sceneId);
+      } else {
+        throw new Error(e);
+      }
+    });
+  }
+
+  // @internal
+  async createLocalEmptyScene(workspaceId: string, sceneId: string): Promise<IotTwinMakerSceneImpl> {
+    const bucketName = await SceneFactoryImpl.iotTwinMaker.getWorkspaceBucketName(workspaceId);
+    return new IotTwinMakerSceneImpl(workspaceId, sceneId, bucketName);
+  }
+
+  public async loadScene(workspaceId: string, sceneId: string): Promise<IotTwinMakerSceneImpl> {
+    const bucketName = await SceneFactoryImpl.iotTwinMaker.getWorkspaceBucketName(workspaceId);
+    const sceneFileContent: string = await SceneFactoryImpl.s3Client.loadSceneFileFrom(bucketName, sceneId);
+    return this.deserializer.deserializeScene(workspaceId, sceneId, sceneFileContent);
+  }
+
+  public async createScene(workspaceId: string, sceneId: string): Promise<IotTwinMakerSceneImpl> {
+    await SceneFactoryImpl.iotTwinMaker.createScene(workspaceId, sceneId);
+    const bucketName = await SceneFactoryImpl.iotTwinMaker.getWorkspaceBucketName(workspaceId);
+    const scene = new IotTwinMakerSceneImpl(workspaceId, sceneId, bucketName);
+    console.log(`Uploading created scene file to ${bucketName}`);
+    await SceneFactoryImpl.s3Client.uploadScene(bucketName, sceneId, this.serializer.serializeScene(scene));
+    return scene;
+  }
+
+  public async deleteScene(workspaceId: string, sceneId: string) {
+    const bucketName = await SceneFactoryImpl.iotTwinMaker.getWorkspaceBucketName(workspaceId);
+    console.log(`Deleting scene ${sceneId}...`);
+    await SceneFactoryImpl.iotTwinMaker.deleteScene(workspaceId, sceneId);
+    await SceneFactoryImpl.s3Client.deleteScene(bucketName, sceneId);
+    console.log('Scene deleted!');
+  }
+
+  public async save(iotTwinMakerScene: IotTwinMakerScene, cesiumAccessToken?: string) {
+    iotTwinMakerScene.selfCheck();
+    const bucketName = await SceneFactoryImpl.iotTwinMaker.getWorkspaceBucketName(iotTwinMakerScene.getWorkspaceId());
+    await this.uploadModelFilesIfNeeded(iotTwinMakerScene, bucketName, cesiumAccessToken);
+    console.log(`Saving scene ${iotTwinMakerScene.getSceneId()}...`);
+    await SceneFactoryImpl.s3Client.uploadScene(
+      bucketName,
+      iotTwinMakerScene.getSceneId(),
+      this.serializer.serializeScene(iotTwinMakerScene as IotTwinMakerSceneImpl),
+    );
+    console.log('Scene saved!');
```

Comment:
(feature request) could be nice to also emit the URL to the scene once saved so the user can quickly open to view it